roles/cloud_connector/tasks/main.yml

```diff
@@ -0,0 +1,63 @@
+---
+- name: Install yggdrasil-worker-forwarder and rhc
+  yum:
+    state: absent
+    name:
+      - rhc
+      - yggdrasil-worker-forwarder
+    disable_plugin: foreman-protector
+
+- name: Create workers directory
+  file:
+    state: directory
+    path: "/etc/rhc/workers"
+    owner: root
+    group: root
+    mode: 0755
+
+- name: Configure rhc-cloud-connector-worker
+  ansible.builtin.template:
+    src: "foreman_rh_cloud.toml.j2"
+    dest: "{{ foreman_cloud_connector_config_file }}"
+    owner: root
+    group: root
+    mode: 0640
+
+- name: Create rhcd worker
+  ansible.builtin.copy:
+    dest: /usr/libexec/rhc/foreman-rh-cloud-worker
+    content: |
+      #!/bin/bash
+
+      CONFIG_FILE="{{ foreman_cloud_connector_config_file }}" exec /usr/libexec/yggdrasil-worker-forwarder
+    owner: root
+    group: root
+    mode: 0755
+
+- name: Ensure rhcd started
+  service:
+    name: rhcd
+    state: started
+
+- name: Read client ID from CN of consumer
+  ansible.builtin.command: openssl x509 -in /etc/pki/consumer/cert.pem -subject -noout
+  register: cert_output
+  when: cert_output is not defined
+
+- name: Set client ID in Satellite
+  ansible.builtin.uri:
```

Comment:
if only we had a setting module in some collection… ;)

Reply:
Somebody should write that

Reply:
_Holding up the irony sign._

https://github.com/theforeman/foreman-ansible-modules/blob/develop/plugins/modules/setting.py